app/services/hashtag_autocomplete_service.rb

```diff
@@ -206,24 +219,20 @@ def search(
 
     return limited_results if limited_results.length >= limit
 
-    # Search the data source for each type, validate and sort results,
-    # and break off from searching more data sources if we reach our limit
+    # Next priority are slugs which start with the search term.
     types_in_priority_order.each do |type|
-      search_results = search_for_type(type, guardian, term, limit - limited_results.length)
-      next if search_results.empty?
-
-      next if !all_data_items_valid?(search_results)
+      limited_results = search_using_condition(limited_results, term, type, limit, SEARCH_CONDITION_STARTS_WITH)
+      top_ranked_type = type if top_ranked_type.nil?
+      break if limited_results.length >= limit
```

Comment:
Why not like this?

```suggestion
      return limited_results if limited_results.length >= limit
```

Reply:
Oh I can't do this because I have to process these at the end to add `::type` to the ones that need it.